soh/soh/OTRGlobals.cpp

```diff
@@ -1718,8 +1712,8 @@ extern "C" int CustomMessage_RetrieveIfExists(GlobalContext* globalCtx) {
             } else {
                 messageEntry = Randomizer_GetGanonHintText();
             }
-        } else if (textId == TEXT_SCRUB_POH || textId == TEXT_SCRUB_STICK_UPGRADE || textId == TEXT_SCRUB_NUT_UPGRADE) {
-            messageEntry = Randomizer_GetScrubMessage(textId);
+        } else if (textId >= 0x9000 && textId <= 0x905F) {
+            messageEntry = Randomizer_GetScrubMessage((textId & ((1 << 8) - 1)));
```

Comment:
i think the `(textId & ((1 << 8) - 1))` logic should move into `GetScrubMessage`, so the `itemPrice` param name makes sense

Reply:
looking at this again, since we only have the textId at this point, it's probably worth just passing in the textId to `GetScrubMessage` and then having commented logic for getting the price out of that textId inside `GetScrubMessage`

Reply:
seeing
```c
// Currently the textID is simply identified by the item price since that is the only thing
// unique to it, later on this will change to identifying by scrubIdentity.randomizerInf
```
makes me think this can be a later problem